huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go

```diff
@@ -611,3 +642,104 @@ func privateCAStatusRefreshFunc(id, region string, cfg *config.Config) resource.
 		return getPrivateCARespBody, status.(string), nil
 	}
 }
+
+func resourcePrivateCAUpdate(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	privateCAClient, err := cfg.NewServiceClient("ccm", region)
+	if err != nil {
+		return diag.Errorf("error creating CCM Client: %s", err)
+	}
+
+	// update tags
+	if d.HasChange("tags") {
+		getTagsHttpUrl := "v1/private-certificate-authorities/{id}/tags"
+		tagsList, err := getTags(d, privateCAClient, getTagsHttpUrl)
```

Comment:
it's better to use d.GetChanges("tags") to delete the old tags and add new tags. please refer to utils.UpdateResourceTags